kernel/src/main/java/org/kframework/parser/inner/kernel/KSyntax2Bison.java

```diff
@@ -215,6 +217,83 @@ private static void encode(Sort sort, StringBuilder sb) {
     sb.append("_");
   }
 
+  private static void appendOverloadCondition(StringBuilder bison, Module module, Production greater, Production lesser, List<Integer> nts) {
+    bison.append("true");
+    for (int i = 0; i < nts.size(); i++) {
+      boolean hasSameSort = lesser.nonterminals().apply(i).sort().equals(greater.nonterminals().apply(i).sort());
+      if (!hasSameSort) {
+        bison.append(" && strncmp($").append(nts.get(i)).append(".nterm->symbol, \"inj{\", 4) == 0 && (false");
+        Sort greaterSort = lesser.nonterminals().apply(i).sort();
+        for (Sort lesserSort : iterable(module.subsorts().elements())) {
+          if (module.subsorts().lessThanEq(lesserSort, greaterSort)) {
+            bison.append(" || strcmp($").append(nts.get(i)).append(".nterm->children[0]->sort, \"");
+            encodeKore(lesserSort, bison);
+            bison.append("\") == 0");
+          }
+        }
+        bison.append(")");
+      }
+    }
+  }
+
+  private static void appendOverloadChecks(StringBuilder bison, Module module, Production greater, List<Integer> nts, boolean hasLocation) {
+    for (Production lesser : iterable(module.overloads().elements())) {
+      if (module.overloads().lessThan(lesser, greater)) {
+        bison.append("  if (");
```

Comment:
This is a repeated `if( ... ) { ... } else` section, which means that only the first `lesser` production that passes the test is processed. Why is it good enough to use only the first one?

Reply:
This is a good catch; I meant to topologically sort the entries, but forgot. This should be fixed now; because of the topological sorting, a lesser overload will always appear before a greater overload, so we will always end up choosing an overload such that there is no lesser overload than the overload we chose that would also satisfy the constraints. As a result, any remaining ambiguities will be real ambiguities.